src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -538,9 +540,12 @@ private async Task<HttpResponseMessage> HandleDeleteHistoryWithFiltersRequestAsy
                 showInput = true;
             }
 
-            if (!TryGetBooleanQueryParameterValue(queryNameValuePairs, ReturnInternalServerErrorOnFailure, out bool returnInternalServerErrorOnFailure))
+            if (returnInternalServerErrorOnFailure == false)

```

Comment:
What if instead of making the default value for the parameter `returnInternalServerErrorOnFailure` false, we made the type `bool?` and default to `null`. Then that allows us to distinguish between the case where the value wasn't provided vs someone intentionally passing in `false`. 

We would still have the guarantee that `returnInternalServerErrorOnFailure` is non-null after this code segment due to the way we set it if the query string doesn't exist.

Reply:
Ah, I like that solution.

Reply:
This would actually make the type that is output from TryGetBooleanQueryParameterValue a compiler error. I would either have to change the output to bool? also or just not make this change.

Reply:
It maybe worth setting a new `bool` variable here for `returnInternalServerErrorOnFailure` (it would need a new name obviously). I still think the nullable input to this method is important to help distinguish between `false` and the default value.

Otherwise you could get some weird unexpected behavior where we look at the query string parameters of the HTTP Trigger being able to override the value explicitly passed into `WaitForCompletionOrCreateCheckStatusResonseAsync()`, which feels incorrect to me.

Reply:
That's fair. I updated this in my latest commit.